infra/prompt-flow/create-prompt-flow.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash -e
+
+SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+echo "Assigning variables from azd env"
+
+while IFS='=' read -r key value; do
+    value=$(echo "$value" | sed 's/^"//' | sed 's/"$//')
+    case "$key" in
+        "ORCHESTRATION_STRATEGY") orchestration_strategy=$value ;;
+        "AZURE_SUBSCRIPTION_ID") subscription_id=$value ;;
+        "AZURE_RESOURCE_GROUP") resource_group=$value ;;
+        "AZURE_ML_WORKSPACE_NAME") aml_workspace=$value ;;
+        "RESOURCE_TOKEN") resource_token=$value ;;
+        "AZURE_OPENAI_RESOURCE") openai_resource=$value ;;
+        "AZURE_OPENAI_EMBEDDING_MODEL") openai_embedding_model=$value ;;
+        "AZURE_SEARCH_SERVICE") search_service=$value ;;
+        "AZURE_SEARCH_INDEX") search_index=$value ;;
+    esac
+done <<EOF
+$(azd env get-values)
+EOF
+
+if [ "$orchestration_strategy" != "prompt_flow" ]; then
+    echo "Orchestration strategy is not prompt_flow, skipping prompt flow creation"
+    exit 0
+fi
+
+if [ -z "$subscription_id" ] || [ -z "$resource_group" ] || [ -z "$aml_workspace" ] || [ -z "$resource_token" ] || [ -z "$openai_resource" ] || [ -z "$openai_embedding_model" ] || [ -z "$search_service" ] || [ -z "$search_index" ]; then
+    echo "AZURE_SUBSCRIPTION_ID, AZURE_RESOURCE_GROUP, AZURE_ML_WORKSPACE_NAME, RESOURCE_TOKEN, AZURE_OPENAI_RESOURCE, AZURE_OPENAI_EMBEDDING_MODEL, AZURE_SEARCH_SERVICE and AZURE_SEARCH_INDEX must be set in azd env"
+    echo "AZURE_SUBSCRIPTION_ID=$subscription_id"
+    echo "AZURE_RESOURCE_GROUP=$resource_group"
+    echo "AZURE_ML_WORKSPACE_NAME=$aml_workspace"
+    echo "RESOURCE_TOKEN=$resource_token"
+    echo "AZURE_OPENAI_RESOURCE=$openai_resource"
+    echo "AZURE_OPENAI_EMBEDDING_MODEL=$openai_embedding_model"
+    echo "AZURE_SEARCH_SERVICE=$search_service"
+    echo "AZURE_SEARCH_INDEX=$search_index"
+    exit 1
+fi
+
+model_name="cwyd-model-${resource_token}"
+endpoint_name="cwyd-endpoint-${resource_token}"
+deployment_name="cwyd-deployment-${resource_token}"
+
+echo "Installing dependencies"
+pip install promptflow[azure]==1.11.0
```

Comment:
Any way we could do this with poetry?

Reply:
We could, but I wanted to only install it if required. I guess we could add another dependency group? Might be overkill for just one dependency though

Reply:
I think, if possible, all our dependencies should be managed using Poetry

Adding another dependency group just with one dependency doesn't seem overkill IMO, we could mark the group as optional, and then here we would just need to have a `poetry install --with ...` command

Reply:
`poetry install --only ...` probably better actually